tests/Features/ReturnTypes/TestCaseExtension.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Features\ReturnTypes;
+
+use App\User;
+use Illuminate\Foundation\Application;
+
+class TestCaseExtension
+{
+    public function testMockMethod(): void
+    {
+        (new TestTestCase())->testMockMethod();
+    }
+
+    public function testPartialMockMethod(): void
+    {
+        (new TestTestCase())->testPartialMockMethod();
+    }
+
+    public function testSpyMethod(): void
+    {
+        (new TestTestCase())->testSpyMethod();
+    }
+}
+
+class TestTestCase extends \Illuminate\Foundation\Testing\TestCase
+{
+    public function testMockMethod(): void
+    {
+        if (method_exists($this, 'mock')) {
```

Comment:
Do we need this `method_exists` checks?

Reply:
I can remove the method_exists check for mock and spy. But for `partialMock ` I have to check if the method exists on `TestCase` because the testsuite of 7.1 install an older version of Laravel where the method doesn't exists. See [travis build](https://travis-ci.org/nunomaduro/larastan/builds/608719982?utm_source=github_status&utm_medium=notification)